### PR TITLE
SITL: added SIM2 message

### PR DESCRIPTION
**Pull request by tridge**
useful for EKF debugging

### CODE REVIEW

**Review by priseborough — APPROVED** [review]

File: libraries/SITL/SIM_Aircraft.cpp [hunk 1]
```diff
@@ -458,6 +459,29 @@ void Aircraft::fill_fdm(struct sitl_fdm &fdm)
         set_speedup(sitl->speedup);
         last_speedup = sitl->speedup;
     }
+
+    // for EKF comparison log relhome pos and velocity at loop rate
+    static uint32_t last_sim2_ms;
+    uint32_t now_ms = AP_HAL::millis();
+    if (now_ms - last_sim2_ms >= AP::scheduler().get_loop_period_s()*1000) {
```
Comment (peterbarker) [inline libraries/SITL/SIM_Aircraft.cpp:466]: Why not just look at `AP::scheduler().ticks()` instead?

File: libraries/SITL/SIM_Aircraft.cpp [hunk 2]
```diff
@@ -458,6 +459,29 @@ void Aircraft::fill_fdm(struct sitl_fdm &fdm)
...
+// @Field: PN: North position from home
+// @Field: PE: East position from home
+// @Field: PD: Down position from home
+// @Field: VN: Velocity north
+// @Field: VE: Velocity east
+// @Field: VD: Velocity down
+        Vector3d pos = get_position_relhome();
+        Vector3f vel = get_velocity_ef();
+        AP::logger().Write("SIM2", "TimeUS,PN,PE,PD,VN,VE,VD",
+                           "Qffffff",
```
Comment (peterbarker) [inline libraries/SITL/SIM_Aircraft.cpp:480]:
```suggestion
                           "Qdddfff",
```

**Review by khancyr — APPROVED** [review]

File: libraries/SITL/SIM_Aircraft.cpp (hunk 1 above)
Comment (tridge) [reply libraries/SITL/SIM_Aircraft.cpp:466]: good idea, done

File: libraries/SITL/SIM_Aircraft.cpp (hunk 2 above)
Comment (tridge) [reply libraries/SITL/SIM_Aircraft.cpp]: done

**Pull request merged**